src/proxy/http/profiles.rs

```diff
@@ -416,7 +416,8 @@ pub mod router {
             let inner = self.inner.make(&target);
             let stack = self.route_layer.clone().service(svc::shared(inner));
 
-            let router = Router::new(
+            // We never want to purge routes from the Profile's router cache
+            let (router, _cache_bg) = Router::new(
```

Comment:
Nit: I think this can just be 
```suggestion
            let (router, _) = Router::new(
```
which will immediately drop the background future